### ISSUE TITLE
TEXT COLOR

### ISSUE THREAD

**Issue by goyakpacking** [WordPress/gutenberg#10225]
Ease of changing Text Color is too cumbersome.

Problem:  We have several sites that use a selected blue font color from the tool bar.  That blue color is used throughout our sites for LINK colors. Theme TwentySeventeen.   When creating a link in TwentySeventeen [to do several changes with ease] the sequence is click the tool bar LINK icon, paste the link in the drop down box, click enter to save the link.  While the text is still highlighted, click the BOLD icon on the tool bar, then click the TEXT COLOR icon drop down, then click one of the specific default chart blue colors [#0000ff] we have been using.

That creates the Link, makes the text bold, and colors it blue, all in a few clicks in the same area.

When we attempt this with Gutenberg, first; the text color change option does not appear in the tool bar above the block we are working on, were the LINK icon is located.  

Off to the right side in some of the options we can change the text color, but the default choice of color in the colored button selection, does not match the TwentySeventeen Theme color defaults.

That makes us create a different colored text, unless diving farther in, with more steps, and steps at different locations on the screen.

Suggestion.  Mimic the two rows of tool bar functions found in the TwentySeventeen Theme.  Further, Mimic the default color chart found there, so they are the same between Gutenberg Editor, and the Classic Editor.   Your Plugin requires too many steps to get the desired effect.  Edit Functions are in different locations [i.e the Tool bar above the Block, and the Options on the left side],  The default color selection mismatching is especially troublesome for someone working on an existing site.

Os Win7 64 Bit
Browser Firefox 62.0 (64-bit)
Gutenberg 3.9.0

**Comment by Soean** [comment]
Related issue: `Need of TextColor for inline formatting` #6080

**Comment by joyously** [comment]
Changing text colors in posts and pages SHOULD be difficult. It should be a rare case, not something you do a lot. 

Changing link color in the theme should be easy. Changing the theme to one with different link colors or with different options to change link colors should be easy. 

And changing the link color in the theme (or changing the theme) should change the whole site, so having unique ones in posts/pages might clash with the new choice and you'd have to revisit each post/page to fix it. This is not best practice or recommended.

**Comment by chris11049** [comment]
Thanks for the judgement on what should or should not be difficult. I disagree entirely with the need to change colors (in links and outside of links), so if someone could actually address the question with an answer I would be thrilled.  I wasn't searching for what someone else deems best practice, that would be a different search.

**Comment by iseulde** [comment]
The plugin [Advanced Rich Text Tools](https://wordpress.org/plugins/advanced-rich-text-tools/) will now add inline text and background color controls. This is plugin territory, we won't add it to WordPress Core at this time.

**Comment by phpbits** [comment]
@goyakpacking @joyously @chris11049  I've also added text highlighting color options on EditorsKit plugin ( https://wordpress.org/plugins/block-options/ ) and I hope this solves this problem. Here's how my implementation works :

![Gutenberg Highlighted Text and Background Colors](https://user-images.githubusercontent.com/3365507/57937370-60a29100-78f8-11e9-8069-06a8ec43259d.gif)

**Comment by goyakpacking** [comment]
Thanks for the update/follow-up. I really appreciate it.

 But for the most part I have stopped using the block editor and went back to using the classic editor.  Especially for the majority of my text content. There are too many things  that don't work seamlessly with already built sites.  I made support tickets but got a few discouraging replies that make me not care to report issues further. Like I was the only one with a particular issue. 

Hoping someday some of these resolved issues are incorporated into the Wordpress core to avoid loading plugins.  I keep trying the Gutenberg block editor from time to time.  But it is not there yet even on some of the most basic functions...from this users perspective anyway. Quite frankly I have used Elementor and find it more user friendly for what I do, if I want to use a block editor.

Many users such as myself just want original features, edits, one click options of original classic color schemes, fonts, in-line edits, left click menu's verses something off on the side of the page, etc to exist.  In addition to new options you are trying to incorporate.

Adding the new options to the original classic toolbar would have been a beautiful thing.

We as users find it cumbersome to be always trying to solve or adapt to a process that was once a simple one click or so.  Or now needing to add a plugin, dealing with updating plug-ins, ending up with pages that don't appear consistent with others on our sites if we don't dig for the process, figuring out multi-steps to replicate our existing pages, or diving into the HTML code when we don't have time to figure it out.  

Just some thoughts and feedback for you. All meant only in a positive constructive sense.

    On Friday, May 17, 2019, 8:11:08 AM PDT, Jeffrey Carandang <notifications@github.com> wrote:  
 
 
@goyakpacking @joyously @chris11049 I've also added text highlighting color options on EditorsKit plugin ( https://wordpress.org/plugins/block-options/ ) and I hope this solves this problem. Here's how my implementation works :
